client/web/src/components/login-toggle.tsx

```diff
@@ -134,50 +134,89 @@ function LoginPopoverContent({
+      >
+        {auth.viewerIdentity ? "Sign in to confirm identity" : "Sign in"}
+      </button>
+    )
+  }, [auth, handleSignInClick])
+
   return (
     <div onMouseEnter={!canConnectOverTS ? checkTSConnection : undefined}>
       <div className="text-black text-sm font-medium leading-tight mb-1">
         {!auth.canManageNode ? "Viewing" : "Managing"}
         {auth.viewerIdentity && ` as ${auth.viewerIdentity.loginName}`}
       </div>
-      {!auth.canManageNode &&
-        (!auth.viewerIdentity || auth.authNeeded === AuthType.tailscale ? (
-          <>
-            <p className="text-neutral-500 text-xs">
-              {auth.viewerIdentity ? (
+      {!auth.canManageNode && (
+        <>
+          {!auth.viewerIdentity ? (
+            // user is not connected over Tailscale
+            <>
+              {!canConnectOverTS ? (
                 <>
-                  To make changes, sign in to confirm your identity. This extra
-                  step helps us keep your device secure.
+                  <p className="text-neutral-500 text-xs">
+                    {!node.ACLAllowsAccess ? (
+                      // Tailnet ACLs don't allow access
+                      <>
+                        The current tailnet policy file does not allow
+                        connecting to this device.
+                      </>
+                    ) : (
+                      // ACLs allow access, but user can't connect
+                      <>
+                        Cannot access this device's Tailscale IP. Make sure you
+                        are connected to your tailnet, and that your policy file
+                        allows access.
+                      </>
+                    )}{" "}
+                    <a
+                      href="https://tailscale.com/s/web-client-connection"
+                      className="text-indigo-700"
+                      target="_blank"
+                      rel="noreferrer"
+                    >
+                      Learn more &rarr;
+                    </a>
+                  </p>
                 </>
               ) : (
+                // User can connect to Tailcale IP; sign in when ready
                 <>
-                  You can see most of this device's details. To make changes,
-                  you need to sign in.
+                  <p className="text-neutral-500 text-xs">
+                    You can see most of this device's details. To make changes,
+                    you need to sign in.
+                  </p>
+                  {signInButton()}
                 </>
               )}
+            </>
+          ) : auth.authNeeded === AuthType.tailscale ? (
+            // User is connected over Tailscale, but needs to complete check mode
+            <>
+              <p className="text-neutral-500 text-xs">
+                To make changes, sign in to confirm your identity. This extra
+                step helps us keep your device secure.
+              </p>
```

Comment:
I think this copy should mirror that on line 195/196. Both cases require check mode I believe.

Reply:
It's slightly different.  On line 195/196, they're using the login client, so we have no identity of the user.  There no avatar, and we just say they are "viewing".  So we just tell them they need to sign in.  Yes, it will end up invoking check mode, but from the user's perspective, it's just a normal sign in flow.

Here, they are already identified, because they're connecting over Tailscale.  It's weird that we need them to login "again", which is why we have the extra language.  This is consistent with what we had before and with Ale's designs.

Reply:
Oh got it - thought it was adding a new copy. This sgtm then.